src/main/java/com/iheartradio/m3u8/MediaPlaylistLineParser.java

```diff
@@ -285,6 +283,31 @@ public void parse(String line, ParseState state) throws ParseException {
         }
     };
 
+    static final IExtTagParser EXT_X_BYTE_RANGE = new IExtTagParser() {
+        private final LineParser lineParser = new MediaPlaylistLineParser(this);
+
+        @Override
+        public String getTag() {
+            return Constants.EXT_X_BYTE_RANGE_TAG;
+        }
+
+        @Override
+        public boolean hasData() {
+            return true;
+        }
+
+        @Override
+        public void parse(String line, ParseState state) throws ParseException {
+            lineParser.parse(line, state);
+            final Matcher matcher = ParseUtil.match(Constants.EXT_X_BYTE_RANGE_PATTERN, line, getTag());
+
+            if (matcher.groupCount() == 3) {
+                String offset = matcher.group(3);
+                state.getMedia().byteRange = new ByteRange(Integer.parseInt(matcher.group(1)), offset == null ? null : Integer.parseInt(offset));
```

Comment:
The spec says the following:
```
If o is not present, the sub-range begins at the next byte following
the sub-range of the previous Media Segment.

If o is not present, a previous Media Segment MUST appear in the
Playlist file and MUST be a sub-range of the same media resource, or
the Media Segment is undefined and the Playlist MUST be rejected.
```
This is a bit unfortunate since it means we will need to add a piece of state to the parsing, but it does mean there is no null case. We need to keep track of the last byte range (which may be nothing) to calculate the offset for the current byte range if no offset is specified. So it would look something like this:
```
offset = group(2);

if (offset is null) {
    if (previous is null) {
        throw;
    } else {
        offset = previous.offset + previous.subRangeLength;
    }
}
```
We're not quite done. Because of this:
```
... a previous Media Segment MUST appear in the
Playlist file and MUST be a sub-range of the same media resource ...
```
we need to clear the previous byte range when changing media sources.